scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -2771,8 +2771,9 @@ public DtlsConnectorConfig build() {
 			}
 
 			if (ecc) {
-				if (!config.supportedSignatureAlgorithms.isEmpty()) {
-					verifySignatureAndHashAlgorithms();
+				if (config.supportedSignatureAlgorithms.isEmpty()) {
+					config.supportedSignatureAlgorithms = SignatureAndHashAlgorithm
+							.getDefaultSignatureAlgorithms(config.certChain);
```

Comment:
In my opinion, for CoAP RFC 7252, [RPK](https://tools.ietf.org/html/rfc7252#section-9.1.3.2) and [x509](https://tools.ietf.org/html/rfc7252#section-9.1.3.3) defines, that each peer MUST support ECDSAwithSHA256. So for compliant implementations, not too much more is required. If for some use-case with "public" CAs more is required, I would just go to add these required sig/hash on demand.

Bascialy, sending more  sig/hash or curves or cipher  suites, end up in larger and larger CLIENT_HELLOs. So I would consider to select a default, which matches the most known/reported use-case and the technical possibilities.

Reply:
This remembers me some discussion about CIPHER.

If I remember well, currently we are not so consistent as we don't really handle default crypto params(cipher_suite, supported group, sign_algo) in a same way.

Personally, I would rather expect that we choose a kind of rule for all crypto params.

Like `default = all system supported one` or `default=only CoAP mandatory one` or something else.

Anyway, I understand well here you choose ECDSAwithSHA256 because it is mandatory for CoAP and RSAwithSHA256 because this is a "common" one.

Reply:
From the discussion of issue #909 

> For now the list is pretty short

Now the curves are added and the sig/hash are added.
For the curves there is some IANA recommendation, that results in a pretty short list.
For sig/hash at least MD5 and SHA1 should be not recommended. Though it's a list of combinations, I'm not sure, how large the rest will get. And my tests (openssl / libcoap interop) showed, that it seems to be more or less ignored for the selection of the certificate (or PSK suite instead). For openssl that may be cause be the "test nature" of these tool  commands, for libcoap I don't know.

I still think, that a small CLIENT_HELLO is important and that for IoT a proper selection of parameter should be used. We had already fragmented CLIENT_HELLOs (scandium fails there. And supporting it would cause state in the stateless verify phase). I would prefer, if other libraries would also provide a proper selection for that.

Reply:
> For now the list is pretty short

I remember well this discussion and I also proposed to only activate the default for CoAP but you didn't want because you wanted to add the LWM2M one too, that I didn't understand.

I'm not be sure to understand the intention about quoting something I said before :thinking: . I don't always agree with my past self, opinions could change.

Anyway, I get your point and I don't fundamentally disagree with you. :)
I think your arguments about having a small CLIENT_HELLO make senses.

My point was just, it would maybe be simple to have the same approach for `signAlgo` `curves` and `ciphersuite`.

And this approach could be : only mandatory by default  + deprecated deactivate by default + easy way to activate all (for demo/tests purpose?) + deactivate deprecated guard. This is just an idea for maybe the 3.0.0 ? it could be pre-introduce in 2.x with an easy way to activate mandatory only with the builder.

Reply:
Sorry, that is a misunderstanding.

I want to express, that the past decision was also taken with the "short list in mind". But now we have three lists.

Reply:
> Sorry, that is a misunderstanding.

no problem, it is clearer now :)

> I want to express, that the past decision was also taken with the "short list in mind". But now we have three lists.

So maybe if the future we should consider the mandatory only by default which should be enough for interoperability.